### ISSUE TITLE
add test for regression in 1.0.20170510165748 & 1.0.20170510151339

### ISSUE THREAD

**Issue by mr-c** [common-workflow-language/cwltool#379]
Fine in v1.0.20170413194156

`Cannot make job: check_adjust() takes exactly 3 arguments (2 given)'`

- [x] fix this
- [ ] add a test for `--cachedir`

here's the diff https://github.com/common-workflow-language/cwltool/compare/0b9086e5cc9b474a064b1a81df752966f143b914...a3c4c31

**Comment by mr-c** [comment]
Triggered via https://github.com/ProteinsWebTeam/ebi-metagenomics-cwl/blob/3d85286/workflows/emg-pipeline-v3-paired-job.yaml

**Comment by manu-chroma** [comment]
``` bash
$ cwl-runner emg-pipeline-v3-paired.cwl                                 18:00:15
/home/manu/github/cwltool/venv/bin/cwl-runner 1.0.20170510165748
Resolved 'emg-pipeline-v3-paired.cwl' to 'file:///home/manu/github/cwltool/workflows/ebi-metagenomics-cwl/workflows/emg-pipeline-v3-paired.cwl'
No handlers could be found for logger "rdflib.term"
usage: emg-pipeline-v3-paired.cwl [-h] --16S_model 16S_MODEL --23S_model
                                  23S_MODEL --5S_model 5S_MODEL
                                  --forward_reads FORWARD_READS
                                  --go_summary_config GO_SUMMARY_CONFIG
                                  --reverse_reads REVERSE_READS --tRNA_model
                                  TRNA_MODEL
                                  [job_order]
emg-pipeline-v3-paired.cwl: error: argument --16S_model is required
```
@mr-c Can you give the exact config for running this worflow so I can reproduce the error? 
The README in their repository is confusing.

**Comment by tetron** [comment]
@mr-c can you post a backtrace with --debug ?

**Comment by tetron** [comment]
@mr-c I think the problem only appears when using `--cachedir`, can you confirm?

**Comment by tetron** [comment]
Possible one line fix in #380

**Comment by mr-c** [comment]
@tetron We are running with `--cachedir` yes

I'll will test the PR tomorrow morning (I'm in Brisbane at the Genomics
Standards Consortium conference)

Pe 16 mai 2017 23:39, "Peter Amstutz" <notifications@github.com> a scris:
